File: casetudy/src/app/customer/model/Customer.ts
import {CustomerType} from "./CustomerType";

export class Customer {
  private _id: number;
  private _code:string;
  private _name: string;
  private _dayOfBirth: string;
  private _gender: string;
  private _card: string;
  private _phone: string;
  private _address: string;
  private _email:string;
  private _customerType: CustomerType;


  constructor(id: number, code: string, name: string, dayOfBirth: string, gender: string, card: string, phone: string, address: string, email: string, customerType: CustomerType) {
    this._id = id;
    this._code = code;
    this._name = name;
    this._dayOfBirth = dayOfBirth;
    this._gender = gender;
    this._card = card;
    this._phone = phone;
    this._address = address;
    this._email = email;
    this._customerType = customerType;
  }

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get name(): string {
    return this._name;
  }

  set name(value: string) {
    this._name = value;
  }

  get dayOfBirth(): string {
    return this._dayOfBirth;
  }

  set dayOfBirth(value: string) {
    this._dayOfBirth = value;
  }

  get gender(): string {
    return this._gender;
  }

  set gender(value: string) {
    this._gender = value;
  }

  get card(): string {
    return this._card;
  }

  set card(value: string) {
    this._card = value;
  }

  get phone(): string {
    return this._phone;
  }

  set phone(value: string) {
    this._phone = value;
  }

  get address(): string {
    return this._address;
  }

  set address(value: string) {
    this._address = value;
  }

  get customerType(): CustomerType {
    return this._customerType;
  }

  set customerType(value: CustomerType) {
    this._customerType = value;
  }

  get code(): string {
    return this._code;
  }

  set code(value: string) {
    this._code = value;
  }

  get email(): string {
    return this._email;
  }

  set email(value: string) {
    this._email = value;
  }
}

File: api_angular/src/main/java/com/example/api_angular/model/Customer.java
package com.example.api_angular.model;

import com.fasterxml.jackson.annotation.JsonBackReference;
import org.hibernate.annotations.Cascade;

import javax.persistence.*;

@Entity
@Table(name = "customer")
public class Customer {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;


    private String name;
    private String code;
    @Column(columnDefinition = "DATE")
    private String dayOfBirth;
    private String gender;
    private String card;
    private String phone;
    private String email;
    private String address;

//    @JsonBackReference
    @ManyToOne
    @JoinColumn(name = "customer_type_id", referencedColumnName = "id")
//    @Cascade(org.hibernate.annotations.CascadeType.ALL)
    private CustomerType customerType;

    public Customer() {
    }

    public Customer(Integer id, String name, String code, String dayOfBirth, String gender, String card, String phone, String email, String address, CustomerType customerType) {
        this.id = id;
        this.name = name;
        this.code = code;
        this.dayOfBirth = dayOfBirth;
        this.gender = gender;
        this.card = card;
        this.phone = phone;
        this.email = email;
        this.address = address;
        this.customerType = customerType;
    }

    public CustomerType getCustomerType() {
        return customerType;
    }

    public void setCustomerType(CustomerType customerType) {
        this.customerType = customerType;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getCode() {
        return code;
    }

    public void setCode(String code) {
        this.code = code;
    }

    public String getDayOfBirth() {
        return dayOfBirth;
    }

    public void setDayOfBirth(String dayOfBirth) {
        this.dayOfBirth = dayOfBirth;
    }

    public String getGender() {
        return gender;
    }

    public void setGender(String gender) {
        this.gender = gender;
    }

    public String getCard() {
        return card;
    }

    public void setCard(String card) {
        this.card = card;
    }

    public String getPhone() {
        return phone;
    }

    public void setPhone(String phone) {
        this.phone = phone;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }


}
